### PR TITLE
ath79: dir-629: use nvmem

### PR DESCRIPTION
**Pull request by neheb**
Userspace handling is deprecated.

ping @robimarko @hauke

### CODE REVIEW

File: target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts [hunk 1]
```diff
@@ -72,6 +72,11 @@
 	};
 };
 
+&eth1 {
+	nvmem-cells = <&macaddr_mfcdata_6a 0>;
+	nvmem-cell-names = "mac-address";
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts:77]: This device uses the external switch chip connected to the `eth0`. It doesn't use eth1, so this patch won't work.

Comment (neheb) [reply target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts:77]: oh so eth1 is not connected to the switch? that's unfortunate.

Comment (DragonBluep) [reply target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts:77]: Yes. QCA955x and QCA9563 don't have built-in switch.

Comment (neheb) [reply target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts:77]: yeah, only two devices have ar8236.

Comment (neheb) [reply target/linux/ath79/dts/qca9558_dlink_dir-629-a1.dts:77]: @Ansuel this switch looks similar to 8327, no?

**Pull request closed without merging**